go.mod

```diff
@@ -1,9 +1,15 @@
 module github.com/bufbuild/protocompile
 
-go 1.16
+go 1.18
```

Comment:
I've added a generic function, to handle reserved ranges for both enums and messages. So that required 1.18.